src/main/java/org/wso2/carbon/securevault/aws/secret/repository/AWSSecretRepository.java

```diff
@@ -0,0 +1,342 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.securevault.aws.secret.repository;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.securevault.aws.common.AWSSecretManagerClient;
+import org.wso2.carbon.securevault.aws.common.AWSVaultUtils;
+import org.wso2.carbon.securevault.aws.exception.AWSVaultException;
+import org.wso2.carbon.securevault.aws.exception.AWSVaultRuntimeException;
+import org.wso2.securevault.CipherFactory;
+import org.wso2.securevault.CipherOperationMode;
+import org.wso2.securevault.DecryptionProvider;
+import org.wso2.securevault.EncodingType;
+import org.wso2.securevault.definition.CipherInformation;
+import org.wso2.securevault.keystore.IdentityKeyStoreWrapper;
+import org.wso2.securevault.keystore.TrustKeyStoreWrapper;
+import org.wso2.securevault.secret.SecretRepository;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
+import software.amazon.awssdk.services.secretsmanager.model.ResourceNotFoundException;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Properties;
+
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.ALGORITHM;
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.CRLF_SANITATION_REGEX;
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.DEFAULT_ALGORITHM;
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.ENCRYPTION_ENABLED;
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.ROOT_PASSWORDS;
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.VERSION_DELIMITER;
+
+/**
+ * AWS secret repository. This class is to facilitate the use of AWS Secrets Manager as an external vault
+ * for the Carbon Secure Vault.
+ */
+public class AWSSecretRepository implements SecretRepository {
+
+    private static final Log log = LogFactory.getLog(AWSSecretRepository.class);
+
+    private SecretRepository parentRepository;
+    // Secret Client used to retrieve secrets from AWS Secrets Manager Vault.
+    private SecretsManagerClient secretsClient;
+    private IdentityKeyStoreWrapper identityKeyStoreWrapper;
+    private TrustKeyStoreWrapper trustKeyStoreWrapper;
+    private DecryptionProvider baseCipher;
+    private boolean encryptionEnabled = false;
```

Comment:
default Boolean value is false. Can you please check

Reply:
Yes its false. But just for clarity, we discussed during a code review to leave it like this.